api/cpp/CMakeLists.txt

```
@@ -191,31 +191,42 @@ if (SLINT_BUILD_RUNTIME)
         endif()
     endif(SLINT_FEATURE_BACKEND_QT)
 
-    if(TARGET Qt::qmake)
+    if(SLINT_FEATURE_BACKEND_QT AND TARGET Qt::qmake)
         set_property(
             TARGET slint-cpp
             APPEND
             PROPERTY CORROSION_ENVIRONMENT_VARIABLES
             QMAKE=$<TARGET_PROPERTY:Qt::qmake,LOCATION>
         )
-        set(SLINT_STYLE_DEFAULT "native")
     else()
         set_property(
             TARGET slint-cpp
             APPEND
             PROPERTY CORROSION_ENVIRONMENT_VARIABLES
             SLINT_NO_QT=1
         )
+    endif()
+
+
+    if(CMAKE_SYSTEM_NAME STREQUAL "Android")
+        set(SLINT_STYLE_DEFAULT "material")
+    elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+        set(SLINT_STYLE_DEFAULT "fluent")
+    elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+        set(SLINT_STYLE_DEFAULT "cupertino")
+    elseif (SLINT_FEATURE_BACKEND_QT AND TARGET Qt::qmake)
+        set(SLINT_STYLE_DEFAULT "qt")
+    else ()
```

I think it's idiomatic to use the pre-defined cmake variables.

```suggestion
    if(ANDROID)
        set(SLINT_STYLE_DEFAULT "material")
    elseif(WIN32)
        set(SLINT_STYLE_DEFAULT "fluent")
    elseif(APPLE)
        set(SLINT_STYLE_DEFAULT "cupertino")
    elseif (SLINT_FEATURE_BACKEND_QT AND TARGET Qt::qmake)
        set(SLINT_STYLE_DEFAULT "qt")
    else ()
```